packages/arb-ts/src/lib/message/L2Transaction.ts

```diff
@@ -0,0 +1,183 @@
+    this.logs = tx.logs
+    this.blockNumber = tx.blockNumber
+    this.confirmations = tx.confirmations
+    this.cumulativeGasUsed = tx.cumulativeGasUsed
+    this.effectiveGasPrice = tx.effectiveGasPrice
+    this.byzantium = tx.byzantium
+    this.type = tx.type
+    this.status = tx.status
+  }
+
+  /**
+   * Get an L2ToL1Transaction events created by this transaction
+   * @returns
+   */
+  public getL2ToL1Events(): L2ToL1TransactionEvent['args'][] {
+    const iface = ArbSys__factory.createInterface()
+    const l2ToL1Event = iface.getEvent('L2ToL1Transaction')
+    const eventTopic = iface.getEventTopic(l2ToL1Event)
+    const logs = this.logs.filter(log => log.topics[0] === eventTopic)
+
+    return logs.map(
+      log => iface.parseLog(log).args as L2ToL1TransactionEvent['args']
+    )
+  }
+
+  private getOutboxAddr(network: L2Network, batchNumber: BigNumber) {
+    // find the outbox where the activation batch number of the next outbox
+    // is greater than the supplied batch
+    const res = Object.entries(network.ethBridge.outboxes)
+      .sort((a, b) => {
+        if (a[1].lt(b[1])) return -1
+        else if (a[1].eq(b[1])) return 0
+        else return 1
+      })
+      .find(
+        (_, index, array) =>
+          array[index + 1] === undefined || array[index + 1][1].gt(batchNumber)
+      )
+
+    if (!res) {
+      throw new ArbTsError(
+        `No outbox found for batch number: ${batchNumber.toString()} on network: ${
+          network.chainID
+        }.`
+      )
+    }
+
+    return res[0]
+  }
+
+  /**
+   * Get any l2-to-l1-messages created by this transaction
+   * @param l2SignerOrProvider
+   */
+  public async getL2ToL1Messages<T extends SignerOrProvider>(
+    l1SignerOrProvider: T,
+    l2Network: L2Network
+  ): Promise<L2ToL1MessageReaderOrWriter<T>[]>
+  public async getL2ToL1Messages<T extends SignerOrProvider>(
+    l1SignerOrProvider: T,
+    l2Network: L2Network
+  ): Promise<L2ToL1MessageReader[] | L2ToL1MessageWriter[]> {
+    const provider = SignerProviderUtils.getProvider(l1SignerOrProvider)
+    if (!provider) throw new Error('Signer not connected to provider.')
+
+    return this.getL2ToL1Events().map(log => {
+      const outboxAddr = this.getOutboxAddr(l2Network, log.batchNumber)
+
+      return L2ToL1Message.fromBatchNumber(
+        l1SignerOrProvider,
+        outboxAddr,
+        log.batchNumber,
+        log.indexInBatch
+      )
+    })
+  }
+
+  /**
+   * Whether the data associated with this transaction has been
+   * made available on L1
+   */
+  public async isDataAvailable(
+    l2Provider: providers.JsonRpcProvider,
+    l1Provider: providers.JsonRpcProvider
+  ): Promise<boolean> {
+    const arbReceipt = await getRawArbTransactionReceipt(
+      l2Provider,
+      this.transactionHash,
+      l1Provider
+    )
+
+    // Data is made available in batches, if the batch info is
+    // available then so is the tx data
+    return !!arbReceipt?.l1InboxBatchInfo
+  }
```

Comment:
I don't believe this is correct (mentioned in previous comment about `l1InboxBatchInfo`)
We need to instead do lookups against the L1 provider.
I agree `isDataAvailable` is the most precise way of describing this, but I don't think its the most intuitive for devs consuming the lib

Reply:
Will need to use the getBatch method from the arb provider

Reply:
I kinda like the name - dyou have any other suggestions? In the comment we mention that this is referring specifically to the data being available on L1. It's also a more general principle rather than referring to batches.